File: code/frontend/Tyr/src/model/SubCategory.ts
import {Module} from "./Module";
import {Category} from "./Category";

/**
 * SubCategory model
 *
 * @param id the id of the subcategory
 * @param name the name of the subcategory
 * @param description the description of the subcategory
 * @param modules the modules of the subcategory
 */
export interface SubCategory {
    id?: number;
    name?: string;
    description?: string;
    category?: Category;
    modules?: Module[];
}

File: code/backend/Odin/src/main/kotlin/pt/isel/odin/model/SubCategory.kt
package pt.isel.odin.model

import com.fasterxml.jackson.annotation.JsonBackReference
import com.fasterxml.jackson.annotation.JsonManagedReference
import jakarta.persistence.CascadeType
import jakarta.persistence.Column
import jakarta.persistence.Entity
import jakarta.persistence.FetchType
import jakarta.persistence.GeneratedValue
import jakarta.persistence.GenerationType
import jakarta.persistence.Id
import jakarta.persistence.JoinColumn
import jakarta.persistence.ManyToOne
import jakarta.persistence.OneToMany
import jakarta.persistence.Table

@Entity
@Table(name = "subcategories")
class SubCategory(
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    val id: Long? = null,

    @ManyToOne(fetch = FetchType.EAGER)
    @JoinColumn(name = "category_id", nullable = false)
    @JsonBackReference
    val category: Category? = null,

    @Column(nullable = false)
    val name: String? = null,

    @OneToMany(mappedBy = "subCategory", cascade = [CascadeType.ALL], fetch = FetchType.EAGER)
    @JsonManagedReference
    val modules: List<Module>? = emptyList()
)
